### PR TITLE
Fixed bug with non-ASCII form_translate output

### PR DESCRIPTION
**Pull request by kaapstorm**

### CODE REVIEW

File: formtranslate/api.py [hunk 1]
```diff
@@ -104,5 +104,5 @@ def form_translate(input_data, operation, version='1.0'):
     stdout, stderr = p.communicate(input_data)
     exit_code = p.wait()
 
-    return ShellResult(stdout=stdout.decode(), stderr=stderr.decode(),
+    return ShellResult(stdout=stdout.decode('utf-8'), stderr=stderr.decode('utf-8'),
```
Comment (czue) [inline formtranslate/api.py]: @dannyroberts or @millerdev to confirm this seems right. i think it is, but will hold off. @kaapstorm assume you were able to test?

**Comment by kaapstorm** [comment]
For some context, this bug was found in my development environment, not in production. It happened after importing an application (specifically [sheel-wvlanka-test](https://www.commcarehq.org/a/sheel-wvlanka-test/)), and selecting one of its translated forms (e.g. "Register Mother").

After specifying a character set when decoding output strings, selecting the form worked as expected.

Here is a traceback:

    Traceback:
    File "/home/nhooper/src/commcare-hq/venv/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
      115.                         response = callback(request, *callback_args, **callback_kwargs)
    File "/home/nhooper/src/commcare-hq/venv/local/lib/python2.7/site-packages/django/views/decorators/http.py" in inner
      41.             return func(request, *args, **kwargs)
    File "/home/nhooper/src/commcare-hq/corehq/apps/domain/decorators.py" in _inner
      77.                     return view_func(req, domain_name, *args, **kwargs)
    File "/home/nhooper/src/commcare-hq/corehq/apps/app_manager/views.py" in view_form
      1117.     return view_generic(req, domain, app_id, module_id, form_id)
    File "./submodules/dimagi-utils-src/dimagi/utils/couch/resource_conflict.py" in new_fn
      24.                     return fn(*args, **kwargs)
    File "/home/nhooper/src/commcare-hq/corehq/apps/app_manager/views.py" in view_generic
      1039.                 "references": app.get_references(),
    File "/home/nhooper/src/commcare-hq/corehq/apps/hqmedia/models.py" in get_references
      689.         return [m.as_dict(lang) for m in self.all_media]
    File "./submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py" in _inner
      11.             return m(*args, **kwargs)
    File "./submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py" in __call__
      127.             value = self.func( *args, **kwargs)
    File "/home/nhooper/src/commcare-hq/corehq/apps/hqmedia/models.py" in all_media
      559.                     f.validate_form()
    File "/home/nhooper/src/commcare-hq/corehq/apps/app_manager/models.py" in validate_form
      667.                                version=self.get_app().application_version)
    File "/home/nhooper/src/commcare-hq/corehq/apps/app_manager/xform.py" in validate_xform
      485.     validation_results = formtranslate.api.validate(source, version=version)
    File "./submodules/formtranslate-src/formtranslate/api.py" in validate
      39.         jar_result = form_translate(input_data, "validate", version=version)
    File "./submodules/formtranslate-src/formtranslate/api.py" in form_translate
      107.     return ShellResult(stdout=stdout.decode(), stderr=stderr.decode(),

    Exception Type: UnicodeDecodeError at /a/sheel-wvlanka-test/apps/view/565c76b28202ff9caf2d4498b0709df9/modules-0/forms-0/
    Exception Value: 'ascii' codec can't decode byte 0xe0 in position 24: ordinal not in range(128)

**Comment by dannyroberts** [comment]
@kaapstorm thanks for the context. @ctsims or @wpride do you know off the top of your head whether all of formtranslate's output is guaranteed to be in UTF-8? Seems like a good bet, but not sure what it does in the case of a form that's not UTF-8 to begin with. This is largely curiosity since it probably doesn't happen much (at all?) in practice.

File: formtranslate/api.py (hunk 1 above)
Comment (kaapstorm) [inline formtranslate/api.py]: @czue, yes. I've added a comment to explain this PR -- how I noticed the bug, and checked I'd resolved it. 

I should also have made it clearer that the error was in the imported app (a form refered to a case property that wasn't found), not on production. I didn't test whether the issue was with stderr or stdout, but I thought it best to specify a character set for both.

**Comment by ctsims** [comment]
I think we do explicitly check, but I'm honestly not 100% sure. Nokia
devices only support a very strict subset (not just UTF8, but UTF8 without
byte order markers), but we use a different lib for formtranslate to speed
things up.

**Pull request merged**